velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -271,9 +274,15 @@ RowVectorPtr NestedLoopJoinProbe::generateOutput() {
   return std::move(output_);
 }
 
-bool NestedLoopJoinProbe::advanceProbeRow() {
+bool NestedLoopJoinProbe::advanceProbe() {
   if (hasProbedAllBuildData()) {
-    ++probeRow_;
+    // For cross joins, if there is a single record on the build side, we return
+    // batches containing all probe records from `input_` at a time.
+    if (isCrossJoin() && buildRowCount_ == 1) {
```

Comment:
We probably don't need `buildRowCount_`, just check `buildVectors_.size() == 1 && buildVectors_[0]->size() == 1`.  We need a helper method for this later to classify the categories.

Reply:
+1. I'll clean up the code a bit as I add another category.